elfi/store.py

```diff
@@ -52,13 +70,29 @@ def __init__(self, outputs=None):
         self.batch_size = None
         self.seed = None
 
+        self.name = name
+        self.prefix = prefix or _get_default_prefix()
+
+        if self.path and os.path.exists(self.path):
+            raise ValueError("A pool with this name already exists in {}. You can use "
+                             "OutputPool.open() to open it.".format(self.prefix))
+
+    @property
+    def output_names(self):
+        return list(self.stores.keys())
+
     @property
     def context_set(self):
         return self.seed is not None and self.batch_size is not None
 
     def set_context(self, context):
```

Comment:
It doesn't take the whole context object, but parts of it. Pools are currently members of contexes, so that would create a two way reference which tend to be cumbersome to maintain. In the future we may combine `ComputationContext` and `Pool` objects but we will have to see.